design/proposals/dbt-delete-merge-schema-branch.md

```diff
@@ -0,0 +1,30 @@
+# `lakectl dbt delete-branch-schema` & `lakectl dbt merge-branch-schema`
+
+## Description
+When using dbt alongside lakeFS, the first thing we would want to do is to create a branch to work on. That branch is created with a dbt schema associated with it (it might have the same name of the branch [by default] or a custom one) that is basically a copy of the schema that was associated with the branch which we branched out from but with different name and location.   
+After running dbt over that branch and conducting tests and validations we would want to either: 
+- delete the branch and the schema associated with it
+- merge them into master/main/production/other branch
+- both.
+
+`lakectl dbt delete-branch-schema` will allow the deletion of a branch and the schema associated with it, or just the schema.  
+`lakectl dbt merge-branch-schema` will allow the merging of a branch to a given branch  and the schema associated with it. 
+### Why do we need it?
+The users that use the dbt command are, well, dbt users. They will interact with the `lakectl` tool in the context of the dbt command (and some basic commands such as `commit` and `merge`).
+Adding the `delete-branch-schema` and `merge-branch-schema` subcommands will make it easier for them to integrate lakeFS into their framework.
+
+## `delete-branch-schema` Usage
+1. `lakectl dbt delete-branch-schema --branch <branch name> --schema <schema name>`: Delete branch and schema with the given names.
+2. `lakectl dbt delete-branch-schema --branch <branch name>`: Delete the branch from the repository that the dbt project's target schema resides at. The command will also try to delete a schema with the same name as the branch (when creating a branch using the `lakectl dbt create-branch-schema` command, the default schema name is the branch name).
+3. `lakectl dbt delete-branch-schema --schema <schema name>`: Delete the given schema.
+4. `lakectl dbt delete-branch-schema`: Failure. Must specify at least `--branch` or `--schema`
+
+## `merge-branch-schema` Usage
```

Comment:
I feel that the combination of branch and schema names creates complexity here. let's try to brainstorm and think about how we can simplify. 
I'm not sure that we need more than the following command at this point
`lakectl dbt merge-branch-schema <source schema name> <destination schema name>`: Merge the source schema to the destination schema.

The reason why I think that this is a good starting point is that we have limited amount of feedback and input from users, and this command supports the request to [merge metadata] (https://github.com/treeverse/lakeFS/issues/2930)

Reply:
I agree that we can start with that (also the delete part can start only with a schema delete), but then again, I do believe that a request will arrive (or questions will get asked) regarding branch merge in the context of this command (or branch delete in the context of the delete command), as it's a convenient method for dbt users.
But I haven't completed my prophecy course on Udemy yet so stay tuned :wink: